heat/core/tests/test_suites/basic_test.py

```diff
@@ -118,36 +128,111 @@ def assert_array_equal(self, heat_array, expected_array):
         heat_func: function
             The function that is to be tested
         numpy_func: function
             The numpy implementation of an equivalent function to test against
+        heat_args: dictionary, optional
+            The keyword arguments that will be passed to the heat function. Array and split function don't need to be
+            specified. Default is {}.
+        numpy_args: dictionary, optional
+            The keyword arguments that will be passed to the numpy function. Array doesn't need to be specified.
+            Default is {}.
         distributed_result: bool, optional
             Specify whether the result of the heat function is distributed across all nodes or all nodes have the full
             result. Default is True.
+        data_types: list of numpy dtypes, optional
+            Tensors with all of these dtypes will be created and tested. Each type must to be a numpy dtype.
+            Default is [numpy.int32, numpy.int64, numpy.float32, numpy.float64]
+        low: int, optional
+            In case one of the data_types has integer types, this is the lower bound for the random values.
+            Default is -10000
+        high: int, optional
+            In case one of the data_types has integer types, this is the upper bound for the random values.
+            Default is 10000
+
+        Raises
+        ------
+        AssertionError if the functions to not perform equally.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import heat as ht
+        >>> self.assert_func_equal((2, 2), ht.exp, np.exp)
+
+        >>> self.assert_func_equal((2, 2), ht.exp, np.log)
+        AssertionError: [...]
+        >>> self.assert_func_equal((1, 3, 5), ht.any, np.any, distributed_result=False)
+
+        >>> heat_args = {'sorted': True, 'axis': 0}
+        >>> numpy_args = {'axis': 0}
+        >>> self.assert_func_equal([5, 5, 5, 5], ht.unique, np.unique, heat_arg=heat_args, numpy_args=numpy_args)
+        """
+        if not isinstance(shape, tuple) and not isinstance(shape, list):
+            raise ValueError(
+                "The shape must be either a list or a tuple but was {}".format(type(shape))
+            )
+
+        for dtype in data_types:
+            tensor = self._create_random_array(shape, dtype=dtype, low=low, high=high)
+            self.assert_func_equal_for_tensor(
+                tensor=tensor,
+                heat_func=heat_func,
+                numpy_func=numpy_func,
+                heat_args=heat_args,
+                numpy_args=numpy_args,
+                distributed_result=distributed_result,
+            )
+
+    def assert_func_equal_for_tensor(
+        self,
+        tensor,
+        heat_func,
+        numpy_func,
+        heat_args=None,
+        numpy_args=None,
+        distributed_result=True,
+    ):
+        """
+        This function will create random tensors of the given shape with different data types.
```

Comment:
these docs are the same as for assert_func_equal. are these functions the same?

Reply:
That was a mistake, I updated the docstring